src/snowflake/snowpark/udtf.py

```diff
@@ -569,86 +641,42 @@ def _do_register_udtf(
         api_call_source: str,
         skip_upload_on_content_match: bool = False,
     ) -> UserDefinedTableFunction:
-        if not isinstance(output_schema, (Iterable, StructType)):
-            raise ValueError(
-                f"'output_schema' must be a list of column names or StructType instance to create a UDTF. Got {type(output_schema)}."
-            )
+
         if isinstance(output_schema, StructType):
             _validate_output_schema_names(output_schema.names)
-        if isinstance(
+            return_type = output_schema
+            output_schema = None
+        elif isinstance(output_schema, PandasDataFrameType):
+            _validate_output_schema_names(output_schema.col_names)
+            return_type = output_schema
+            output_schema = None
+        elif isinstance(
             output_schema, Iterable
         ):  # with column names instead of StructType. Read type hints to infer column types.
+            # can we refactor this block to be in process_registration_inputs?
```

Comment:
nit: maybe add a TODO?

Reply:
I refactored already, lemme remove this comment haha.